src/parse-file.ts

```diff
@@ -79,30 +88,46 @@ export default function parseFile(
         }
 
         markers[id] = {
-            checksum,
+            contentChecksum:
+                content == null ? NoChecksum : calcChecksum(content),
+            selfChecksum: calcChecksum([...(content ?? []), relativeFilePath]),
             targets,
             commentStart,
             commentEnd,
         };
     };
 
     const referencedFiles: Array<string> = [];
-    const normalizeFileRef = (fileRef: string) => {
-        const normalizedFileInfo = getNormalizedTargetFileInfo(
-            rootPath,
-            fileRef,
-        );
-        if (!readOnly && normalizedFileInfo.exists) {
-            referencedFiles.push(normalizedFileInfo.file);
+    const normalizeTargetPath = (targetRef: string) => {
+        const targetInfo = getNormalizedPathInfo(rootPath, targetRef);
+        if (!readOnly && targetInfo.exists && targetInfo.type === "local") {
+            referencedFiles.push(targetInfo.path);
         }
-        return normalizedFileInfo;
+        return targetInfo;
     };
 
     return new Promise<FileParseResult>((resolve, reject) => {
         try {
             const markerParser = new MarkerParser(
-                normalizeFileRef,
-                addMarker,
+                normalizeTargetPath,
+                (
+                    id: string,
+                    content: ReadonlyArray<string> | undefined,
+                    targets: Targets,
+                    commentStart: string,
+                    commentEnd: string,
+                ) =>
+                    addMarker(
+                        id,
+                        // We need the normalized path of the file we're
+                        // processing to include it in the selfChecksum so
+                        // that it's the same across different OSes.
+                        normalizeSeparators(path.relative(rootPath, file)),
```

Comment:
note: This was the easiest way I found to get the file path of the file being parsed over to the addMarker function so that it can be used in the selfChecksum value.